src/components/CalendarDay.jsx

```diff
@@ -64,7 +67,7 @@ export default class CalendarDay extends React.Component {
         onMouseLeave={e => this.onDayMouseLeave(day, e)}
         onClick={e => this.onDayClick(day, e)}
       >
-        {day.format('D')}
+        {renderDay ? renderDay(day) : day.format('D')}
```

Comment:
@bamieh we already discussed that here: https://github.com/airbnb/react-dates/pull/307#pullrequestreview-20665548